Mcst/BasicMcst.cs

```diff
@@ -85,7 +102,7 @@ private void Backpropagation(IMcstGame<TMove> game, Node<TMove> node, Stack<TMov
         {
             node.Update(result);
             node = node.Parent;
-            if (node != null)  // Ensure we don't undo the move of the root node as it has no parent
+            if (node != null && moveHistory.Count() > 0)  // Ensure we don't undo the move of the root node as it has no parent
```

Comment:
Tak się teraz zastanawiam czy nam wgl jest po coś potrzebne to moveHistory? Bo w pierwszej wersji chyba było ale teraz i tak mamy clone w RunIteration() więc może można by to usunąć?